synapse/util/manhole.py

```diff
@@ -135,9 +139,8 @@ def showtraceback(self):
         """
         sys.last_type, sys.last_value, last_tb = ei = sys.exc_info()
         sys.last_traceback = last_tb
-        try:
-            # We remove the first stack item because it is our own code.
-            lines = traceback.format_exception(ei[0], ei[1], last_tb.tb_next)
-            self.write("".join(lines))
-        finally:
-            last_tb = ei = None
```

Comment:
Why do you think it is safe to remove this?

Reply:
`last_tb` and `ei` are dead.

can you see a reason why it's like that? It really needs a comment if there's a reason :/

Reply:
the original was cargo-culted from https://github.com/python/cpython/blob/main/Lib/code.py#L150, if that helps at all.

Reply:
> `last_tb` and `ei` are dead.

Technically not, an exception raised here keeps the local frame around to allow inspection (which is how things like Sentry can report the local variables of each frame in the stack). 

I don't know if or why that is an important detail (maybe to prevent recursion if the exception thrown goes back via this function? Or just preventing circular references?), but I'm loathed to remove something like that unless we know why it was put there. (Obligatory reference to [Chesterton's fence](https://en.wikipedia.org/wiki/G._K._Chesterton#Chesterton's_fence))

Reply:
> Technically not, an exception raised here keeps the local frame around to allow inspection (which is how things like Sentry can report the local variables of each frame in the stack).

I don't follow this. This method doesn't raise an exception, so how would `last_db` and `ei` be retained in a stack frame?

I think it's worth digging into the history of the cpython code. It looks to me like those lines were introduced before @reivilibre was born, and at the time there was more code where it might have been useful that the traceback be gc-able.

Reply:
I was assuming `self.write` could throw, otherwise there's also a question over why it uses `try/finally`